src/projects/projects.service.ts

```diff
@@ -312,22 +312,24 @@ export class ProjectService {
     }
   }
 
-  async changeProjectStatus(
+  async modifyProjectStatus(
     companyId: string,
     id: string,
     status: ProjectStatus,
   ) {
     try {
-      return this.postgresService.project.update({
-        where: {
-          id,
-          companyId,
-          deletedAt: null,
-        },
-        data: {
-          status,
-        },
-      });
+      // Raw SQL is used here because Prisma does not support updating JSON fields
+      return this.postgresService.$executeRaw`
+      UPDATE "Project"
+      SET 
+        "status" = ${status}::"ProjectStatus",
+        "meta" = "meta"::jsonb || CASE 
+          WHEN ${status} = 'Completed' THEN jsonb_build_object('endedAt', NOW())
+        END
+      WHERE 
+        "id" = ${id} 
+        AND "companyId" = ${companyId} 
+        AND "deletedAt" IS NULL;`;
```

Comment:
either you have a built-in query for that purpose or you can utilize the new features of prisma 
https://www.prisma.io/docs/orm/prisma-client/special-fields-and-types/working-with-json-fields#advanced-example-update-a-nested-json-key-value